cmd/fpga_plugin/fpga_plugin.go

```diff
@@ -172,17 +175,34 @@ type devicePlugin struct {
 	ignoreAfuIDs       bool
 	ignoreEmptyRegions bool
 	annotationValue    string
+
+	scanTicker *time.Ticker
+	scanDone   chan bool
 }
 
 // newDevicePlugin returns new instance of devicePlugin
-func newDevicePlugin(mode string) (*devicePlugin, error) {
-	if _, err := os.Stat(sysfsDirectoryOPAE); os.IsNotExist(err) {
-		if _, err = os.Stat(sysfsDirectoryDFL); os.IsNotExist(err) {
-			return nil, fmt.Errorf("kernel driver is not loaded: neither %s nor %s sysfs entry exists", sysfsDirectoryOPAE, sysfsDirectoryDFL)
+func newDevicePlugin(mode string, rootPath string) (*devicePlugin, error) {
+	dp, err := func(mode string, rootPath string) (*devicePlugin, error) {
```

Comment:
What's the purpose of this lambda called once? Is it to fight cyclomatic complexity? Then a good-old named function would suffice I guess.

Reply:
The purpose is to keep code in one place. I thought that moving 7 lines of code to another function would look worse than that. Cyclomatic complexity increased, but not that much as the code passed CI check. I can move it though if you insist.

Reply:
Wouldn't the code be in one place without being wrapped into this function then?

Reply:
No, it wouldn't. It would be splited in 2 functions in 2 visually and logically different places.

Reply:
but again, it's not a big deal to split it if needed. I like it less that way, but I want this PR to be merged more than that :) Just let me know if you think it should be splited.

Reply:
Well, I'm just struggling to understand why the variant below can't work :) I don't see any need for splitting.

```golang
func newDevicePlugin(mode string, rootPath string) (*devicePlugin, error) {
    var dp *devicePlugin
    var err error

    sysfsPathOPAE := path.Join(rootPath, sysfsDirectoryOPAE)
    devfsPath := path.Join(rootPath, devfsDirectory)
    if _, err := os.Stat(sysfsPathOPAE); os.IsNotExist(err) {
        sysfsPathDFL := path.Join(rootPath, sysfsDirectoryDFL)
        if _, err: = os.Stat(sysfsPathDFL); os.IsNotExist(err) {
            return nil, errors.Errorf("kernel driver is not loaded: neither %s nor %s sysfs entry exists", sysfsPathOPAE, sysfsPathDFL)
        }
        dp, err = newDevicePluginDFL(sysfsPathDFL, devfsPath, mode)
    } else {
        dp, err = newDevicePluginOPAE(sysfsPathOPAE, devfsPath, mode)
    }

    if err != nil {
        return nil, err
    }

    ...
}
```

Reply:
Thank you! That's better than either of previously discusses changes. Done. Please review again.